server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleQueryFactory.java

```diff
@@ -118,4 +102,47 @@ public RuleQuery createRuleQuery(DbSession dbSession, Request request) {
     }
     return query;
   }
+
+  private void setProfile(DbSession dbSession, RuleQuery query, Request request) {
+    String profileUuid = request.param(PARAM_QPROFILE);
+    if (profileUuid == null) {
+      return;
+    }
+    QProfileDto profileOptional = dbClient.qualityProfileDao().selectByUuid(dbSession, profileUuid);
+    QProfileDto profile = checkFound(profileOptional, "The specified qualityProfile '%s' does not exist", profileUuid);
+    query.setQProfile(profile);
+  }
+
+  private void setOrganization(DbSession dbSession, RuleQuery query, Request request) {
+    String organizationKey = request.param(PARAM_ORGANIZATION);
+    QProfileDto profile = query.getQProfile();
+    if (profile == null) {
+      query.setOrganization(wsSupport.getOrganizationByKey(dbSession, organizationKey));
+      return;
+    }
+    OrganizationDto organization = checkFoundWithOptional(dbClient.organizationDao().selectByUuid(dbSession, profile.getOrganizationUuid()), "No organization with UUID ",
+      profile.getOrganizationUuid());
+    if (organizationKey != null) {
+      OrganizationDto inputOrganization = checkFoundWithOptional(dbClient.organizationDao().selectByKey(dbSession, organizationKey), "No organization with key '%s'",
+        organizationKey);
+      checkArgument(organization.getUuid().equals(inputOrganization.getUuid()),
+        format("The specified quality profile '%s' is not part of the specified organization '%s'", profile.getKee(), organizationKey));
+    }
+    query.setOrganization(organization);
+  }
+
+  private void setCompareToProfile(DbSession dbSession, RuleQuery query, Request request) {
+    String compareToProfileUuid = request.param(PARAM_COMPARE_TO_PROFILE);
+    if (compareToProfileUuid == null) {
+      return;
+    }
+    // TODO fail if qProfile param is not set
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)